federationtypes.go

```diff
@@ -317,23 +316,33 @@ func (r RespSendJoin) MarshalJSON() ([]byte, error) {
 	// of a two element list where the first element is the constant integer 200.
 	// (This protocol oddity is the result of a typo in the synapse matrix
 	//  server, and is preserved to maintain compatibility.)
-	return json.Marshal([]interface{}{200, respSendJoinFields(r)})
+	return json.Marshal(respSendJoinFields{
+		StateEvents: r.StateEvents,
+		AuthEvents:  r.AuthEvents,
+		Origin:      r.Origin,
+	})
 }
 
 // UnmarshalJSON implements json.Unmarshaller
 func (r *RespSendJoin) UnmarshalJSON(data []byte) error {
-	var tuple []RawJSON
+	/*var tuple []RawJSON
```

Comment:
git is a thing: remove dead code, don't comment it

Reply:
Was sure I'd removed this but apparently had reset since :-) fixed.